frontend/packages/dev-console/src/components/import/git/SourceSecretSelector.tsx

```diff
@@ -1,46 +1,84 @@
 import * as React from 'react';
-import { FormGroup, ControlLabel, FormControl, HelpBlock } from 'patternfly-react';
+import { FormGroup, ControlLabel } from 'patternfly-react';
+import { useFormikContext, FormikValues, useField } from 'formik';
+import { createModalLauncher } from '@console/internal/components/factory/modal';
+import {
+  withSecretForm,
+  SourceSecretForm,
+  SecretTypeAbstraction,
+} from '@console/internal/components/secrets/create-secret';
 import SourceSecretDropdown from '../../dropdown/SourceSecretDropdown';
-import CreateSourceSecret, { SecretType } from './CreateSourceSecret';
 
 const CREATE_SOURCE_SECRET = 'create-source-secret';
 
 interface SourceSecretSelectorProps {
-  namespace?: string;
-  sourceSecret: string;
-  selectedKey: string;
-  secretCredentials: { [key: string]: string };
-  onChange: (
-    name: string,
-    key: string,
-    authType?: string,
-    credentials?: { [key: string]: string },
-  ) => void;
+  namespace: string;
 }
 
-const SourceSecretSelector: React.FC<SourceSecretSelectorProps> = ({
-  sourceSecret,
+interface CreateSourceSecretModalProps {
+  cancel: (e: MouseEvent) => void;
```

Comment:
Use the `on` prefix for callback props.
```suggestion
  onCancel: (e: MouseEvent) => void;
```

Reply:
@christianvogt  `CreateSourceSecretModalProps` should  satisfy the constraint 'ModalComponentProps`. `cancel` prop  is required in type 'ModalComponentProps'